src/main/ruby/truffleruby/core/io.rb

```diff
@@ -1648,6 +1650,23 @@ def printf(fmt, *args)
     write sprintf(fmt, *args)
   end
 
+  def timeout
+    @timeout
+  end
+
+  def timeout=(new_timeout)
+    if Primitive.nil?(new_timeout)
+      self.nonblock = false
+    else
+      self.nonblock = true
+
+      # For validation.
+      Truffle::KernelOperations.convert_duration_to_seconds(new_timeout)
+    end
+
+    @timeout = new_timeout
+  end
```

Comment:
Is there a source where you found this? Asking because poking CRuby:

```ruby
STDOUT.send("timeout=")
(irb):12:in `timeout=': wrong number of arguments (given 0, expected 1) (ArgumentError)
```

Reply:
Will create an issue.

> Is there a source where you found this?

No, there is no source. I just was trying to break it in different ways.